command/update/command.go

```diff
@@ -201,44 +201,53 @@ func (c *Command) execute() error {
 	}
 
 	// Listen to OS signals issued by the Kubernetes scheduler.
-	listener := make(chan os.Signal, 2)
-	signal.Notify(listener, syscall.SIGTERM, syscall.SIGKILL)
+	signalCh := make(chan os.Signal, 2)
+	signal.Notify(signalCh, syscall.SIGTERM, syscall.SIGKILL)
 
 	c.logger.Log("debug", "waiting for termination signal")
 
-	s := <-listener
+	s := <-signalCh
 
-	c.logger.Log("debug", fmt.Sprintf("received termination signal: %#v (%s)\n", s, s))
+	c.logger.Log("debug", fmt.Sprintf("received termination signal: %#v (%s)", s, s))
+
+	errCh := make(chan error)
 
 	// Use the updater to actually delete the endpoints identified by the provided
 	// flags.
 	go func() {
+		defer close(errCh)
+
 		action := func() error {
 			err := newUpdater.Delete(f.Kubernetes.Cluster.Namespace, f.Kubernetes.Cluster.Service, podInfos)
 			if err != nil {
 				return microerror.Mask(err)
 			}
-
 			return nil
 		}
 
 		err := backoff.Retry(action, backoff.NewExponentialBackOff())
 		if err != nil {
+			errCh <- microerror.Mask(err)
 			c.logger.Log("error", fmt.Sprintf("%#v", microerror.Mask(err)))
```

Comment:
This would log the error twice. Lets remove it here and only log below.

Reply:
Done, thanks